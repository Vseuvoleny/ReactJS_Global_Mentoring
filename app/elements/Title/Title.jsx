import React from "react";
import "./Title.module.scss";

const Title = () => {
  return (
    <h1 className="title">
      <span>netflix</span>
      <span>roulette</span>
    </h1>
  );
};

export default Title;
