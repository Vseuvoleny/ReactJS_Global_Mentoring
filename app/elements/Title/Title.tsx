import React from "react";
import classes from "./Title.module.scss";

const Title = () => {
  return (
    <h1 className={classes.title}>
      <span>netflix</span>
      <span>roulette</span>
    </h1>
  );
};

export default Title;
