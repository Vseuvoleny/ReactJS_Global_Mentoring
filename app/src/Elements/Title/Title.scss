.title {
  color: #f65261;
  font-size: 20px;
  line-height: 24px;
  font-weight: 500;
  span:first-of-type {
    font-weight: 900;
  }
}
