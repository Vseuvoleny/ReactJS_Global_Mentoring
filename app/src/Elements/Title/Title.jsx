import React, { memo } from "react";
import "./Title.scss";

const Title = memo(() => {
  return (
    <h1 className="title">
      <span>netflix</span>
      <span>roulette</span>
    </h1>
  );
});

export default Title;
