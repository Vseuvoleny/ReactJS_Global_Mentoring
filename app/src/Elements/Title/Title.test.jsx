import React from "react";
import Title from "./Title";
import { render } from "@testing-library/react";

describe("Title component", () => {
  test("should render Title component", () => {
    const { container, getByText } = render(<Title />);
    expect(getByText("netflix")).toBeInTheDocument();
    expect(getByText("roulette")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
