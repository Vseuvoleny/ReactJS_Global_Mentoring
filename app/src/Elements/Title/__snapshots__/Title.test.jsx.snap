// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render Title component 1`] = `
<div>
  <h1
    class="title"
  >
    <span>
      netflix
    </span>
    <span>
      roulette
    </span>
  </h1>
</div>
`;
